patterns/MVC/README.md

```diff
@@ -0,0 +1,299 @@
+
+Sheldon, Robert. “What Is Model-View-Controller (MVC)?: Definition from TechTarget.” WhatIs, TechTarget, 12 Sept. 2023, [techtarget.com](www.techtarget.com/whatis/definition/model-view-controller-MVC#:~:text=In%20programming%2C%20model%2Dview%2D,a%20specific%20set%20of%20tasks)
+
+# Visual representation of MVC architecture
+
+![image](https://github.com/YogeshManni/pattern-library/assets/29475936/96056260-a719-4651-bee0-685ce6390af8)
+
+sadikarahmantanisha, "the-mvc-architecture", medium.com,19 Sept. 2023, [MVC architecture](https://medium.com/@sadikarahmantanisha/the-mvc-architecture-97d47e071eb2)
+
+---
+
+## Advantages of MVC
+
+- **Separation of Concerns**: MVC separates the application into distinct components, making it easier to understand, maintain, and modify.
+- **Modular Design**: Each component of MVC can be developed and tested independently, allowing for greater flexibility and reusability.
+- **Scalability**: MVC facilitates the scalability of the application by allowing developers to add new features or modify existing ones without affecting other parts of the application.
+
+---
+
+# MVC Architecture in Todo Task List Application
+
+For this issue, I will be explaining MVC architecture by writing code for to-do task list application. The Tech stack that I will be using is
+
+```
+Mongodb and Javascript - For Model part (Database)
+React - For View Part
+Node.js - For Controller part
+```
+
+# Folder Structure
+
+    MVC
+    ├── controllers                   # Controllers directory
+    |   |
+    |   |---authentication.js         # Controller for authentication-related requests
+    |   |---category.js               # Controller for category-related requests
+    |   |---server.js                 # Main entry point for backend server
+    |   |---tasks.js                  # Controller for todo task-related requests
+    |
+    ├── models                        # Models directory
+    │   ├── authenticationModel.js    # Model for authentication operations
+    │   ├── categoryModel.js          # Model for category operations
+    │   ├── tasksModel.js             # Model for todo task operations
+    │
+    │---views                          # Views directory
+    |    ├── api.js                    # Service for making API requests
+    |    └── App.js                    # Main component for frontend React application
+    |    |---Login.js                  # View component for login page
+    |    |---NewTask.js                # View component for creating a new task
+    |    |---Register.js               # View component for registration page
+    |    |---Todo.js                   # View component for main todo task list page
+    |    |---TodoItem.js               # View component for rendering an individual todo task
+    |    │---TodoList.js               # Component for rendering the list of todo tasks
+
```

Comment:
Minor, but you have some inconsistency in formatting here. Use a consistent double dash (rather than triple dash) to indicate hierarchical file relationships:
```
-- folder
|  |--file  //prefer this
|  |---file  //don't do this
```

Notice as well that I've made the above a code block by wrapping it in triple backticks (```). You can also accomplish this by tabbing in all text that you wish to be rendered as code (which is why this works with your first line, `MVC`, tabbed in). Using the triple backticks is clearer.